remote_control.py

```diff
@@ -12,7 +12,7 @@
 
 
 class RC():
-    def __init__(self, timeout=120):
+    def __init__(self, timeout=120, joystick=ControllerResource()):

```

Comment:
This is very much a no-no with python. default arguments are evaluated when the function is parsed, not when it is run, which possibly isn't what you want. The common idiom is to default your argument to `None`, and then testing for that in the `init()` method and setting the default argument there.

In this case, I'd advise raising an exception and halting the challenge if the joystick is not provided - no joystick, no RC.

Reply:
ok, I noticed some funny behaviour with this.